### ISSUE TITLE
Empty file import error on Linux aarch64

### ISSUE THREAD

**Issue by todd-heidesch-work-account** [greenplum-db/gpdb#16250]
## Bug Report
Copy from empty file or stdin reports incorrect insertion record count, when destination table is distributed replicated. The issue mostly bothers me because it may be an exploitable buffer overflow.

### Greenplum version or build
7.0.0 beta 4
### OS version and uname -a
Amazon Linux 2023
Linux gpnc 6.1.38-59.109.amzn2023.aarch64 #1 SMP Tue Jul 11 23:51:18 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux

### Installation information ( pg_config )
BINDIR = /usr/local/gpdb/bin
DOCDIR = /usr/local/gpdb/share/doc/postgresql
HTMLDIR = /usr/local/gpdb/share/doc/postgresql
INCLUDEDIR = /usr/local/gpdb/include
PKGINCLUDEDIR = /usr/local/gpdb/include/postgresql
INCLUDEDIR-SERVER = /usr/local/gpdb/include/postgresql/server
LIBDIR = /usr/local/gpdb/lib
PKGLIBDIR = /usr/local/gpdb/lib/postgresql
LOCALEDIR = /usr/local/gpdb/share/locale
MANDIR = /usr/local/gpdb/share/man
SHAREDIR = /usr/local/gpdb/share/postgresql
SYSCONFDIR = /usr/local/gpdb/etc/postgresql
PGXS = /usr/local/gpdb/lib/postgresql/pgxs/src/makefiles/pgxs.mk
CONFIGURE = '--with-perl' '--with-python' '--with-libxml' '--with-gssapi' '--prefix=/usr/local/gpdb' '--with-uuid=ossp' 'PYTHON=/usr/bin/python3.9'
CC = gcc
CPPFLAGS = -D_GNU_SOURCE -I/usr/include/libxml2
CFLAGS = -Wall -Wmissing-prototypes -Wpointer-arith -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-unused-but-set-variable -Werror=implicit-fallthrough=3 -Wno-format-truncation -Wno-stringop-truncation -O3  -Werror=uninitialized -Werror=implicit-function-declaration
CFLAGS_SL = -fPIC
LDFLAGS = -Wl,--as-needed -Wl,-rpath,'/usr/local/gpdb/lib',--enable-new-dtags
LDFLAGS_EX = 
LDFLAGS_SL = 
LIBS = -lpgcommon -lpgport -lxerces-c -lbz2 -lxml2 -lrt -lgssapi_krb5 -lz -lreadline -lcrypt -lm  -lcurl -lzstd 
VERSION = PostgreSQL 12.12

### Expected behavior
COPY 0
### Actual behavior
COPY 18446744073709551615

### Step to reproduce the behavior

touch /home/admin/emptyfile.txt

drop table if exists tablex_replicated;
create table tablex_replicated (x text) with (appendoptimized=true, compresstype=zstd, compresslevel=3, blocksize=1048576, orientation=column) distributed replicated;

psql -d db -c "copy tablex_replicated from '/home/gpadmin/emptyfile.txt';"
COPY 18446744073709551615

drop table if exists tablex_randomly;
create table tablex_randomly (x text) with (appendoptimized=true, compresstype=zstd, compresslevel=3, blocksize=1048576, orientation=column) distributed randomly;
psql -d db -c "copy tablex_randomly from '/home/gpadmin/emptyfile.txt';"
COPY 0

Running the command through JDBC (DBeaver) causes the client to error and disconnect:
copy tablex_replicated from '/home/gpadmin/emptyfile.txt';
SQL Error [08006]: Unable to parse the count in command completion tag: COPY 18446744073709551615.
...but this seems likely to be a driver issue.

**Comment by ashwinstar** [comment]
Sure thanks @todd-heidesch-work-account for report. We will fix it.

**Comment by hyongtao-db** [comment]
I am so sorry for that I introduce this bug in this commit https://github.com/greenplum-db/gpdb/pull/14989 . 
I forget to add a empty file to test my code change. I will fix it right now.

Sorry again.

**Comment by hyongtao-db** [comment]
I think this patch will make it OK.
``` diff
diff --git a/src/backend/cdb/cdbcopy.c b/src/backend/cdb/cdbcopy.c
index a05f31b..66fc588 100644
--- a/src/backend/cdb/cdbcopy.c
+++ b/src/backend/cdb/cdbcopy.c
@@ -644,7 +644,7 @@ cdbCopyEndInternal(CdbCopy *c, char *abort_msg,
                        }
 
                        /* in SREH mode, check if this seg rejected (how many) rows */
-                       if (res->numRejected > 0)
+                       if (res->numRejected >= 0)
                        {
                                segment_rows_rejected = res->numRejected;
                                /* 
@@ -668,7 +668,7 @@ cdbCopyEndInternal(CdbCopy *c, char *abort_msg,
                         * When COPY FROM, need to calculate the number of this
                         * segment's completed rows
                         */
-                       if (res->numCompleted > 0)
+                       if (res->numCompleted >= 0)
                        {
                                segment_rows_completed = res->numCompleted;
                                /* 
```

**Comment by hyongtao-db** [comment]
Fix in #16451 , Close it. Thanks a lot.
